src/AzureAppConfigurationImpl.ts

```diff
@@ -99,19 +274,36 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         return key;
     }
 
-    #enableRequestTracing() {
-        this.#correlationContextHeader = createCorrelationContextHeader(this.#options);
-    }
-
-    #customHeaders() {
+    #customHeaders(requestType: RequestType) {
         if (!this.#requestTracingEnabled) {
             return undefined;
         }
 
         const headers = {};
-        headers[CorrelationContextHeaderName] = this.#correlationContextHeader;
+        headers[CorrelationContextHeaderName] = createCorrelationContextHeader(this.#options, requestType);
         return headers;
     }
+
+    async #getConfigurationSettingWithTrace(configurationSettingId: ConfigurationSettingId, customOptions?: GetConfigurationSettingOptions): Promise<GetConfigurationSettingResponse | undefined> {
+        let response: GetConfigurationSettingResponse | undefined;
+        try {
+            const options = {...customOptions ?? {}};
+            if (this.#requestTracingEnabled) {
+                const requestType = this.#isInitialLoadCompleted ? RequestType.Watch : RequestType.Startup;
+                options.requestOptions = {
+                    customHeaders: this.#customHeaders(requestType)

```

Comment:
Since we added this new variable `isInitialLoadCompleted`, we can simplify wherever `requestType` is passed. We only need to check `isInitialLoadCompleted` while calling `createCorrelationContextHeader` now.

Reply:
Or maybe we can get rid of `#customHeaders` and directly call `createCorrelationContextHeader` wherever we want to get custom headers.

Reply:
good idea. I can start with changing `#customHeaders(requestType: RequestType)` to `#customHeaders()`, and check isInitialLoadCompleted in its implmentation. 
`createCorrelationContextHeader(...)` is a util function assembling correlation context header, which I would prefer to keep it as is, accepting request type as an input.

Reply:
I'll also change `createCorrelationContextHeader(options, requestType)` to `createCorrelationContextHeader(options, isInitialLoadCompleted)` as suggested, which makes sense.